cmd/preview/deploy.go

```diff
@@ -233,33 +233,37 @@ func waitToBeDeployed(ctx context.Context, name, namespace string, timeout time.
 	}
 }
 
-func waitForResourcesToBeRunning(ctx context.Context, name, namespace string, t time.Duration) error {
+func waitForResourcesToBeRunning(ctx context.Context, name, namespace string, timeout time.Duration) error {
 	areAllRunning := false
 
 	ticker := time.NewTicker(5 * time.Second)
-	timeout := time.Now().Add(t)
+	to := time.NewTicker(timeout)
 	errors := make(map[string]int)
-	for time.Now().Before(timeout) {
-		<-ticker.C
-		resourceStatus, err := okteto.GetResourcesStatusFromPreview(ctx, namespace)
-		if err != nil {
-			return err
-		}
-		areAllRunning = true
-		for name, status := range resourceStatus {
-			if status != "running" {
-				areAllRunning = false
+
+	for {
+		select {
+		case <-to.C:
+			return fmt.Errorf("preview environment '%s' didn't finish after %s", name, timeout.String())
+		case <-ticker.C:
+			resourceStatus, err := okteto.GetResourcesStatusFromPreview(ctx, namespace)
+			if err != nil {
+				return err
 			}
-			if status == "error" {
-				errors[name] = 1
+			areAllRunning = true
+			for name, status := range resourceStatus {
+				if status != "running" {
+					areAllRunning = false
+				}
+				if status == "error" {
+					errors[name] = 1
+				}
+			}
+			if len(errors) > 0 {
+				return fmt.Errorf("preview environment '%s' deployed with resource errors", name)
```

Comment:
```suggestion
				return fmt.Errorf("preview environment '%s' deployed with errors", name)
```

Reply:
do you think it would be helpful if we added a hint with the URL of the preview env?